### PR TITLE
feat: Add second verbosity level to `getrawtransaction` rpc call

### PR DESCRIPTION
**Pull request by str0g**
Include deque fix compilation on archlinux. Rest is required for email notification after refactor.

### CODE REVIEW

File: src/rpc/rawtransaction.cpp [hunk 1]
```diff
@@ -185,9 +231,9 @@ static UniValue getrawtransaction(const JSONRPCRequest& request)
     }
 
     // Accept either a bool (true) or a num (>=1) to indicate verbose output.
```
Comment (rrozek) [inline src/rpc/rawtransaction.cpp:233]: update comment

File: src/rpc/rawtransaction.cpp [hunk 2]
```diff
@@ -185,9 +231,9 @@ static UniValue getrawtransaction(const JSONRPCRequest& request)
...
-    bool fVerbose = false;
+    int fVerbose = 0;
```
Comment (rrozek) [inline src/rpc/rawtransaction.cpp:234]: verbosityLevel would be more accurate

File: src/rpc/rawtransaction.cpp [hunk 3]
```diff
@@ -252,6 +298,49 @@ static UniValue getrawtransaction(const JSONRPCRequest& request)
...
+                }
+            }
+        }
+
+        int64_t sum_vout = 0;
+        for (auto& vout : tx->vout) {
+            sum_vout += vout.nValue;
+        }
+
+        result.pushKV("fee", ValueFromAmount(sum_vin - sum_vout));
```
Comment (rrozek) [inline src/rpc/rawtransaction.cpp]: doesn't it require special handling for coinbase transactions? (will result in fee < 0
sth around this lines?
`sum_vin == 0 ? 0 : sum_vin - sum_vout`

**Review by rrozek — CHANGES_REQUESTED** [review]

File: src/rpc/rawtransaction.cpp (hunk 3 above)
Comment (str0g) [reply src/rpc/rawtransaction.cpp]: Nope, its not point of interest.

File: src/rpc/rawtransaction.cpp (hunk 2 above)
Comment (str0g) [reply src/rpc/rawtransaction.cpp:234]: Original implementation was capable to use false, 0  or true any number expect 0.
Current change give option 2 but rest states the same.
I expected that this should be backward compatible.

File: src/rpc/rawtransaction.cpp (hunk 1 above)
Comment (rrozek) [reply src/rpc/rawtransaction.cpp:233]: ah, indeed that comment is stupid.

Comment (rrozek) [reply src/rpc/rawtransaction.cpp:233]: i mean mine, not the one in code. it should stay as it is

**Review by rrozek — DISMISSED** [review]

**Review by rrozek555 — DISMISSED** [review]

File: src/httpserver.cpp [hunk 4]
```diff
@@ -19,6 +19,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <deque>
```
Comment (cloudjrojek) [inline src/httpserver.cpp:22]: ?

File: src/rpc/rawtransaction.cpp (hunk 3 above)
Comment (mtwaro) [reply src/rpc/rawtransaction.cpp]: Coinbase case should be handled here. In current form this call will return non-zero fee for coinbase transactions, which would be incorrect.

**Review by mtwaro — APPROVED** [review]

**Pull request merged**